packages/runtime-vapor/src/apiSetup.ts

```diff
@@ -0,0 +1,72 @@
+import { proxyRefs } from '@vue/reactivity'
+import { type Data, isArray, isFunction, isObject } from '@vue/shared'
+import { type Block, fragmentKey } from './apiRender'
+import {
+  type ComponentInternalInstance,
+  componentKey,
+  setCurrentInstance,
+} from './component'
+import { fallThroughAttrs } from './componentAttrs'
+import type { EmitFn } from './componentEmits'
+import type { EmitsOptions } from 'vue'
+
+export type SetupContext<E = EmitsOptions> = E extends any
+  ? {
+      attrs: Data
+      emit: EmitFn<E>
+      expose: (exposed?: Record<string, any>) => void
+      // TODO slots
+    }
+  : never
+
+export function setupComponent(
```

Comment:
`setupComponent` is not a public API. Please move back to `apiRender.ts`, and move `createSetupContext` to `component.ts`